File: python/grains/grains.py
def on_square(integer_number):

    if integer_number not in range(1, 65):
        raise ValueError("That's not a chessboard square number!")
    
    return 2**(integer_number - 1)


def total_after(integer_number):
    
    if integer_number not in range(1, 65):
        raise ValueError("That's not a chessboard square number!")
    
    total = 0

    for i in range(1, integer_number + 1):
        total += on_square(i)

    return total
        

File: c/grains/src/grains.c
#include "grains.h"

uint64_t square(int sq)
{
        if (sq < 1 || sq > 64)
                return 0; 

        uint64_t total = 1; 

        for (int i = 1; i < sq; i++) {
                total *= 2;
        }

        return total; 
}

uint64_t total()
{
        return 0xFFFFFFFFFFFFFFFF; 
}